ci-operator/step-registry/ipi/conf/azure/resourcegroup/ipi-conf-azure-resourcegroup-commands.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
```

Comment:
We shouldn't be installing in steps. You will need to use an image that has yq. This could mean editing the upi image to install yq.

Reply:
Yes, I was actually just noticing this in some other test work that I was doing. One issue that I have with this, however, is agreement on a `yq` version. The version used here is deprecated, and the current versions are not compatible. I would love to see us move to the current versions globally, but I don't know the extent of where `yq` is being used throughout the steps. I suppose we could add `yq` to the image and existing uses of `yq` could still pull the deprecated version until they have a chance to migrate.

Reply:
> which is becoming a bit of a dumping ground

I was also thinking of this. It does seem strange to use the UPI image for IPI installs, too.

Reply:
> but otherwise doing a curl install like this is far too risky to do in the upi image IMHO

We are already doing a bunch of curls in the UPI image. Other options for putting `yq` in the image are
1. running `go get github.com/mikefarah/yq/v4`, or
2. copying from the `mikefarah/yq` image.

Reply:
> I was also thinking of this. It does seem strange to use the UPI image for IPI installs, too.
Right but it is just the config setup that uses the UPI image for the tooling. The actual install runs on the main installer image.

>We are already doing a bunch of curls in the UPI image. Other options for putting yq in the image are
Ok I'm fine with the curling. If github goes down we have bigger problems. Just a little hesitant because breaking the UPI image will break a lot of tests, but I walk back my statement this is "far too risky." `curl`ing or installing by any well known method seems good to me.

Reply:
Apparently, we should be creating images in https://github.com/openshift/release/tree/master/clusters/app.ci/supplemental-ci-images to use instead of building the CI image with every job.

Reply:
> Just a little hesitant because breaking the UPI image will break a lot of tests

Funny you should mention that.
```
 Downloading https://files.pythonhosted.org/packages/53/7f/55721ad0501a9076dbc354cc8c63ffc2d6f1ef360f49ad0fbcce19d68538/pip-20.3.4.tar.gz#sha256=6773934e5f5fc3eaa8c5a44949b5b924fc122daa0a8aa9f80c835b4ca2a543fc
error: https://files.pythonhosted.org/packages/53/7f/55721ad0501a9076dbc354cc8c63ffc2d6f1ef360f49ad0fbcce19d68538/pip-20.3.4.tar.gz#sha256=6773934e5f5fc3eaa8c5a44949b5b924fc122daa0a8aa9f80c835b4ca2a543fc hostname 'files.pythonhosted.org' doesn't match either of 'r.ssl.fastly.net', '*.catchpoint.com', '*.cnn.io', '*.dollarshaveclub.com', '*.eater.com', '*.fastly.picmonkey.com', '*.files.saymedia-content.com', '*.ft.com', '*.meetupstatic.com', '*.nfl.com', '*.pagar.me', '*.picmonkey.com', '*.realself.com', '*.sbnation.com', '*.shakr.com', '*.streamable.com', '*.surfly.com', '*.theverge.com', '*.thrillist.com', '*.vox-cdn.com', '*.vox.com', '*.voxmedia.com', 'eater.com', 'ft.com', 'i.gse.io', 'picmonkey.com', 'realself.com', 'static.wixstatic.com', 'streamable.com', 'surfly.com', 'theverge.com', 'vox-cdn.com', 'vox.com', 'www.joyent.com'
error: build error: error building at STEP "RUN easy_install 'pip<21'": error while running runtime: exit status 1 
```